var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -389,6 +389,10 @@ def install(self, spec, prefix):
                 raise SpackException(
                     'The lldb variant requires the `+clang` variant')
 
+        platform = self.spec.architecture.platform
+        if spec.satisfies('@4.0.0:') and platform == 'linux':
```

Comment:
The suggested change doesn't seem to work, I pushed up something that did work for me and is a bit cleaner than before.